diagnostics/collect-networking-logs.ps1

```diff
@@ -0,0 +1,217 @@
+
+$wslconfig = "$env:USERPROFILE/.wslconfig"
+if (Test-Path $wslconfig)
+{
+    Copy-Item $wslconfig $folder
+}
+
+# Collect Linux network state before the repro
+& wsl.exe -e ./networking.sh 2>&1 > $folder/linux_network_configuration_before.log
+
+if ($RestartWslReproMode)
+{
+    # The WSL HNS network is created once per boot. Resetting it to collect network creation logs.
+    Get-HnsNetwork | Where-Object {$_.Name -eq 'WSL'} | Remove-HnsNetwork
+
+    # Stop WSL.
+    net.exe stop WslService
+    if(-not $?)
+    {
+        net.exe stop LxssManager
+    }
+}
+
+# Start logging.
+$logProfile = ".\wsl_networking.wprp"
+$wprOutputLog = "$folder/wpr.txt"
+
+wpr.exe -start $logProfile -filemode 2>&1 >> $wprOutputLog
+if ($LastExitCode -Ne 0)
+{
+    Write-Host -ForegroundColor Yellow "Log collection failed to start (exit code: $LastExitCode), trying to reset it."
+    wpr.exe -cancel 2>&1 >> $wprOutputLog
+
+    wpr.exe -start $logProfile -filemode 2>&1 >> $wprOutputLog
+    if ($LastExitCode -Ne 0)
+    {
+        Write-Host -ForegroundColor Red "Couldn't start log collection (exitCode: $LastExitCode)"
+    }
+}
+
+# Start packet capture using pktmon
+pktmon start -c --flags 0x1A --file-name "$folder/pktmon.etl" | out-null
+
+# Start WFP capture
+netsh wfp capture start file="$folder/wfpdiag.cab"
+
+# Start tcpdump. Using a try/catch as tcpdump might not be installed
+$tcpdumpProcess = $null
+try
+{
+    $tcpdumpProcess = Start-Process wsl.exe -ArgumentList "-u root tcpdump -n -i any > $folder/tcpdump.log" -PassThru
+}
+catch {}
+
+try
+{
+    Write-Host -NoNewLine -ForegroundColor Green "Log collection is running. Please reproduce the problem and press any key to save the logs."
+
+    $KeysToIgnore =
+          16,  # Shift (left or right)
+          17,  # Ctrl (left or right)
+          18,  # Alt (left or right)
+          20,  # Caps lock
+          91,  # Windows key (left)
+          92,  # Windows key (right)
+          93,  # Menu key
+          144, # Num lock
+          145, # Scroll lock
+          166, # Back
+          167, # Forward
+          168, # Refresh
+          169, # Stop
+          170, # Search
+          171, # Favorites
+          172, # Start/Home
+          173, # Mute
+          174, # Volume Down
+          175, # Volume Up
+          176, # Next Track
+          177, # Previous Track
+          178, # Stop Media
+          179, # Play
+          180, # Mail
+          181, # Select Media
+          182, # Application 1
+          183  # Application 2
+
+    $Key = $null
+    while ($Key -Eq $null -Or $Key.VirtualKeyCode -Eq $null -Or $KeysToIgnore -Contains $Key.VirtualKeyCode)
+    {
+        $Key = $Host.UI.RawUI.ReadKey('NoEcho,IncludeKeyDown')
+    }
+
+    Write-Host "`nSaving logs..."
+}
+finally
+{
+    try
+    {
+        wsl.exe -u root killall tcpdump
+        if ($tcpdumpProcess -ne $null)
+        {
+            Wait-Process -InputObject $tcpdumpProcess

```

Comment:
nit: We might want a timeout of let's say 10 seconds here, to make sure we don't wait forever in case something went wrong